securedrop_client/storage.py

```diff
@@ -274,8 +312,12 @@ def update_sources(
     """
     local_sources_by_uuid = {s.uuid: s for s in local_sources}
     for source in remote_sources:
-        if source.uuid in local_sources_by_uuid:
+        if source.uuid in skip_uuids_deleted_source:
```

Comment:
The first thing we check is if the source has just been locally deleted. If so, we can just skip all of the logic below